### PR TITLE
split node_modules dirs into their own child treecaches

### PR DESCRIPTION
**Pull request by jdhollen**
**Related issues**: N/A

### CODE REVIEW

File: server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go [hunk 1]
```diff
@@ -456,12 +458,52 @@ func makeTreeCacheActionResult(blob *rspb.ResourceName) ([]byte, error) {
...
+			eg.Go(func() error {
+				childBuf, err := proto.Marshal(childCache)
+				if err != nil {
+					return err
+				}
+				childBlob, err := makeTreeCacheDigest(treeCachePointer.GetDigestFunction(), childBuf)
+				if err != nil {
+					return err
+				}
+				if err := s.cache.Set(egCtx, childBlob.ToProto(), childBuf); err != nil {
```
Comment (tylerwilliams) [inline server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go:491]: nit: wanna add a `if not s.cache.Contains()` checks before any CAS sets we do?

**Review by tylerwilliams — APPROVED** [review]
LGTM

(i think you're already doing this but making a test that triggers the split-out-dir case would be nice)

File: server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go (hunk 1 above)
Comment (jdhollen) [reply server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go]: Yup, done

**Comment by jdhollen** [comment]
> LGTM
> 
> (i think you're already doing this but making a test that triggers the split-out-dir case would be nice)

Added a test--the splitting itself is invisible at the call layer, but I've set the flags in the test case such that splitting happens.

**Pull request merged**